### PR TITLE
chore(ci): add junit reporter in e2e test workflows

### PR DESCRIPTION
**Pull request by odockal**
### What does this PR do?
Junit reporter is now part of e2e jobs in various workflows (e2e tests). Significantly improves test result visibility and readability without a need to download artifacts.
### What issues does this PR fix or reference?
#8847 

### How to test this PR?
Check E2E test workflow run results page on github checks.

- [ ] Tests are covering the bug fix or the new feature

### CODE REVIEW

File: .github/workflows/pr-check.yaml [hunk 1]
```diff
@@ -295,6 +295,9 @@ jobs:
     if: ${{ contains(github.event.pull_request.labels.*.name, 'area/ci') || !github.event.pull_request.draft }}
     name: smoke e2e tests
     runs-on: ubuntu-24.04
+    permissions:
+      contents: read
+      checks: write
```
Comment (benoitf) [inline .github/workflows/pr-check.yaml]: question: I don't understand how you can ask permissions to write in the PR check job where users have no rights ?

Comment (odockal) [reply .github/workflows/pr-check.yaml]: Let me check if it is needed.

Comment (odockal) [reply .github/workflows/pr-check.yaml]: Seems like with `annotate_only` it still works fine. removing `permissions` from all changes.

**Review by danivilla9 — APPROVED** [review]
LGTM, the results are way easier to read this way

**Review by benoitf — APPROVED** [review]

**Pull request merged**